src/routes/transactions/mappers/transactions-history.mapper.ts

```diff
@@ -188,54 +188,41 @@ export class TransactionsHistoryMapper {
     onlyTrusted: boolean,
   ): Promise<TransactionItem[]> {
     const limitedTransfers = transfers.slice(0, this.maxNestedTransfers);
-    const result: TransactionItem[] = [];
 
-    for (const transfer of limitedTransfers) {
-      const nestedTransaction = await this.transferMapper.mapTransfer(
-        chainId,
-        transfer,
-        safe,
-      );
-
-      const transferWithValue = this.mapZeroValueTransfer(nestedTransaction);
-      // If we do not have a transfer with value, we do not add it to the result
-      if (!transferWithValue) continue;
+    const nestedTransactions = await Promise.all(
+      limitedTransfers.map((transfer) =>
+        this.transferMapper.mapTransfer(chainId, transfer, safe),
+      ),
+    );
 
-      const trustedTransfer = onlyTrusted
-        ? this.mapTrustedTransfer(transferWithValue)
-        : transferWithValue;
+    return nestedTransactions
+      .filter((nestedTransaction): boolean => {
+        // If we do not have a transfer with value, we do not add it to the result
+        if (!this.isTransferWithVaklue(nestedTransaction)) return false;
 
-      if (!trustedTransfer) continue;
-      result.push(new TransactionItem(nestedTransaction));
-    }
-    return result;
+        return !onlyTrusted || this.isTrustedTransfer(nestedTransaction);
+      })
+      .map((nestedTransaction) => new TransactionItem(nestedTransaction));
   }
 
   /**
-   * Returns the transaction if it is an ERC20 transfer with value.
-   * Returns Null otherwise.
+   * Returns true if it is an ERC20 transfer with value.
+   * Returns false otherwise.
    *
    * @private
    */
-  private mapZeroValueTransfer(transaction: Transaction): Transaction | null {
-    if (!isTransferTransactionInfo(transaction.txInfo)) return transaction;
-    if (!isErc20Transfer(transaction.txInfo.transferInfo)) return transaction;
+  private isTransferWithVaklue(transaction: Transaction): boolean {
+    if (!isTransferTransactionInfo(transaction.txInfo)) return true;
+    if (!isErc20Transfer(transaction.txInfo.transferInfo)) return true;
 
-    if (transaction.txInfo.transferInfo.value === '0') return null;
-    return transaction;
+    return transaction.txInfo.transferInfo.value !== '0';
```

Comment:
This might be a bug (even in the previous version) as a value of `'00'` should probably be considered as zero too.

Reply:
I've changed it to compare the numerical value in 15489f36e5464501c0356a4652eae757dd04c396.